src/main/java/bio/terra/service/resourcemanagement/BufferService.java

```diff
@@ -0,0 +1,131 @@
+     * projects. Note that this is configured once per Workspace Manager instance (both the instance
+     * of RBS to use and which pool) so no configuration happens here.
+     *
+     * @return PoolInfo
+     */
+    public PoolInfo getPoolInfo() {
+        try {
+            BufferApi bufferApi = bufferApi(bufferServiceConfiguration.getInstanceUrl());
+            PoolInfo info = bufferApi.getPoolInfo(bufferServiceConfiguration.getPoolId());
+            logger.info(
+                "Retrieved pool {} on Buffer Service instance {}",
+                bufferServiceConfiguration.getPoolId(),
+                bufferServiceConfiguration.getInstanceUrl());
+            return info;
+        } catch (IOException e) {
+            throw new BufferServiceAuthorizationException(
+                "Error reading or parsing credentials file", e.getCause());
+        } catch (ApiException e) {
+            if (e.getCode() == HttpStatus.UNAUTHORIZED.value()) {
+                throw new BufferServiceAuthorizationException(
+                    "Not authorized to access Buffer Service", e.getCause());
+            } else {
+                throw new BufferServiceAPIException(e);
+            }
+        }
+    }
+
+    /**
+     * Retrieve a single resource from the Buffer Service. The instance and pool are already
+     * configured.
+     *
+     * @param requestBody
+     * @return ResourceInfo
+     */
+    public ResourceInfo handoutResource(HandoutRequestBody requestBody) {
+        try {
+            BufferApi bufferApi = bufferApi(bufferServiceConfiguration.getInstanceUrl());
+            ResourceInfo info =
+                bufferApi.handoutResource(requestBody, bufferServiceConfiguration.getPoolId());
+            logger.info(
+                "Retrieved resource from pool {} on Buffer Service instance {}",
+                bufferServiceConfiguration.getPoolId(),
+                bufferServiceConfiguration.getInstanceUrl());
+            return info;
+        } catch (IOException e) {
+            throw new BufferServiceAuthorizationException(
+                "Error reading or parsing credentials file", e.getCause());
+        } catch (ApiException e) {
+            if (e.getCode() == HttpStatus.UNAUTHORIZED.value()) {
+                throw new BufferServiceAuthorizationException(
+                    "Not authorized to access Buffer Service", e.getCause());
+            } else {
+                throw new BufferServiceAPIException(e);
+            }
+        }
+    }
+
+    public RepositoryStatusModelSystems status() {
+        UnauthenticatedApi unauthenticatedApi = new UnauthenticatedApi(getUnAuthApiClient());
+        try {
+            SystemStatus status = unauthenticatedApi.serviceStatus();
+            Map<String, SystemStatusSystems> subsystemStatusMap =
+                status.getSystems();
+            return new RepositoryStatusModelSystems()
+                .ok(status.isOk())
+                .critical(false)
```

Comment:
Is the plan to make this critical once we depend on it to create projects?

Reply:
Yeah, and we could actually use `rbs.enabled` instead of hard-coding this to `false`